lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -656,6 +674,9 @@ def render_figures(code, code_path, output_dir, output_base, context,
             for format, dpi in formats:
                 try:
                     figman.canvas.figure.savefig(img.filename(format), dpi=dpi)
+                    if config.plot_preserve_dir and outname:
+                      print("Preserving '{0}' into '{1}'".format(img.filename(format), config.plot_preserve_dir))
```

Comment:
Not reviewing, but I don't think you want `print` in here do you?  I'd do `_log.info` or `_log.debug`?

Reply:
Yes, that's probably what I want, but I didn't know about them. I'll update.

Reply:
Actually, I don't see `_log` as a variable in the `plot_directive.py` namespace, so I'm not sure how to make that adjustment.

Reply:
If its not already in this module, then up at the top: 
```python
# ... other imports
import logging

# ... other imports

_log = logging.getLogger(__name__)

```

Reply:
Thanks: got that up.